src/app/layout/carpooling/carpooling.component.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { CarpoolingComponent } from './carpooling.component';
+import { ADDRESS_SERVICE_TOKEN, AddressServiceInterface } from 'src/app/interface/other';
+import { MatCardModule } from '@angular/material/card';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { of } from 'rxjs';
+import { By } from '@angular/platform-browser';
+
+describe('CarpoolingComponent', () => {
+  let component: CarpoolingComponent;
+  let fixture: ComponentFixture<CarpoolingComponent>;
+  let spyAddressService: jasmine.SpyObj<AddressServiceInterface>;
+
+  beforeEach(() => {
+    spyAddressService = jasmine.createSpyObj('AddressServiceInterface', ['search', 'find', 'matchingSchoolDeparture']);
+    TestBed.configureTestingModule({
+      declarations: [CarpoolingComponent],
+      imports: [
+        HttpClientTestingModule,
+        MatCardModule
+      ],
+      providers: [
+        {provide: ADDRESS_SERVICE_TOKEN, useValue: spyAddressService},
+      ]
+    });
+    fixture = TestBed.createComponent(CarpoolingComponent);
+    component = fixture.componentInstance;
+    spyAddressService.$schoolList = [
+      {lat: 48.9757551, lon: 2.559337, name: 'IUT de Tremblay-en-France'}
+    ];
+    spyAddressService.matchingSchoolDeparture.and.callFake((lat: number, lon: number): string | undefined => {
+      for (let school of spyAddressService.$schoolList) {
+        if (school.lat === lat && school.lon === lon) {
+          return school.name;
+        }
+      }
+      return undefined;
+    });
```

Comment:
```suggestion
spyAddressService.matchingSchoolDeparture.and.callFake((lat: number, lon: number): string | undefined => 
  spyAddressService.$schoolList.find(school => school.lat === lat && school.lon === lon)?.name
);
```
A tester (j'suis pas sûr)